File: C_Quests.cpp
#include <bits/stdc++.h>
#define int long long
#define ab adityabelgaonkar
using namespace std;

int32_t main()
{
    ios_base::sync_with_stdio(false); cin.tie(NULL);
    
    int t; cin >> t;
    while(t--)
    {
        int n, k; cin >> n >> k;
        vector<int> a(n), b(n);
        vector<int> c;
        int maxel = b[0];

        for(auto &i : a) cin >> i;
        for(int i = 1; i < n; ++i) a[i] += a[i - 1];
        for(auto &i : b) cin >> i;
        for(int i = 0; i < n; ++i)
        {
            maxel = max(maxel, b[i]);
            c.push_back(maxel);
        }

        int right = min(k - 1, n - 1);
        int ans = INT_MIN;
        while(right >= 0)
        {
            int temp = a[right];
            if(k > right - 1)
            {
                temp += ((k - right - 1) * c[right]);
            }

            ans = max(ans, temp);
            right--;
        }

        // for(int i : a) cout << i << ' '; cout << '\n';
        // for(int i : c) cout << i << ' '; cout << '\n';

        cout << ans << '\n';
    }

    return 0;
}

File: C_Quests.py
import sys
input = sys.stdin.read

# submitting to compare slowness not cheating!

def main():
    data = input().split()
    index = 0
    
    t = int(data[index])
    index += 1
    
    results = []
    
    for _ in range(t):
        n = int(data[index])
        k = int(data[index + 1])
        index += 2
        
        a = list(map(int, data[index:index + n]))
        index += n
        b = list(map(int, data[index:index + n]))
        index += n
        
        c = []
        maxel = b[0]
        
        for i in range(1, n):
            a[i] += a[i - 1]
        
        for i in range(n):
            maxel = max(maxel, b[i])
            c.append(maxel)
        
        right = min(k - 1, n - 1)
        ans = -float('inf')
        
        while right >= 0:
            temp = a[right]
            if k > right - 1:
                temp += ((k - right - 1) * c[right])
            
            ans = max(ans, temp)
            right -= 1
        
        results.append(str(ans))
    
    print('\n'.join(results))

if __name__ == "__main__":
    main()
